contracts/OriginsAdmin.sol

```diff
@@ -0,0 +1,225 @@
+	event VerifierAdded(address indexed _initiator, address _newVerifier);
+
+	/**
+	 * @notice Emitted when a verifier is removed.
+	 * @param _initiator The one who initiates this event.
+	 * @param _removedVerifier The verifier who has been removed.
+	 */
+	event VerifierRemoved(address indexed _initiator, address _removedVerifier);
+
+	/* Modifiers */
+
+	/**
+	 * @dev Throws if called by any account other than the owner.
+	 */
+	modifier onlyOwner() {
+		require(isOwner[msg.sender], "OriginsAdmin: Only owner can call this function.");
+		_;
+	}
+
+	/**
+	 * @dev Throws if called by any account other than the verifier.
+	 */
+	modifier onlyVerifier() {
+		require(isVerifier[msg.sender], "OriginsAdmin: Only verifier can call this function.");
+		_;
+	}
+
+	/* Functions */
+
+	/**
+	 * @dev Initializes the contract, setting the deployer as the initial owner.
+	 * @param _owners The owners list.
+	 */
+	constructor(address[] memory _owners) internal {
+		uint256 len = _owners.length;
+		for (uint256 index = 0; index < len; index++) {
+			require(!isOwner[_owners[index]], "OriginsAdmin: Each owner can be added only once.");
+			isOwner[_owners[index]] = true;
+			owners.push(_owners[index]);
+			emit OwnerAdded(msg.sender, _owners[index]);
+		}
+	}
+
+	/**
+	 * @notice The function to add a new owner.
+	 * @param _newOwner The address of the new owner.
+	 * @dev Only callable by an Owner.
+	 */
+	function addOwner(address _newOwner) public onlyOwner {
+		_addOwner(_newOwner);
+	}
+
+	/**
+	 * @notice The function to remove an owner.
+	 * @param _ownerToRemove The address of the owner which should be removed.
+	 * @dev Only callable by an Owner.
+	 */
+	function removeOwner(address _ownerToRemove) public onlyOwner {
+		_removeOwner(_ownerToRemove);
+	}
+
+	/**
+	 * @notice The function to add a new verifier.
+	 * @param _newVerifier The address of the new verifier.
+	 * @dev Only callable by an Owner.
+	 */
+	function addVerifier(address _newVerifier) public onlyOwner {
+		_addVerifier(_newVerifier);
+	}
+
+	/**
+	 * @notice The function to remove an verifier.
+	 * @param _verifierToRemove The address of the verifier which should be removed.
+	 * @dev Only callable by an Owner.
+	 */
+	function removeVerifier(address _verifierToRemove) public onlyOwner {
+		_removeVerifier(_verifierToRemove);
+	}
+
+	/* Internal Functions */
+
+	/**
+	 * @notice The internal function to add a new owner.
+	 * @param _newOwner The address of the new owner.
+	 */
+	function _addOwner(address _newOwner) internal {
+		require(_newOwner != address(0), "OriginsAdmin: Invalid Address.");
+		require(!isOwner[_newOwner], "OriginsAdmin: Address is already an owner.");
+		isOwner[_newOwner] = true;
+		owners.push(_newOwner);
+
+		emit OwnerAdded(msg.sender, _newOwner);
+	}
+
+	/**
+	 * @notice The internal function to remove an owner.
+	 * @param _ownerToRemove The address of the owner which should be removed.
+	 */
+	function _removeOwner(address _ownerToRemove) internal {
```

Comment:
In a general case, there should not be more than 2 admins, so this should not be taking much gas. Maybe a new function to do this and call `_removeOwner`, and edit the `_removeOwner` to use an index as well?

Reply:
In the future, instead of `Ownable`, I might use the `AccessControl` which in this case looks more useful due to different ownership levels like Owner, Verifier (and upcoming seller roles, etc).